### PR TITLE
Release preview version 2024-07-31 of HybridCompute

### PR DESCRIPTION
**Pull request by yaotongms**
# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

### CODE REVIEW

File: sdk/hybridcompute/Azure.ResourceManager.HybridCompute/src/Generated/GatewayData.cs [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+using Azure.ResourceManager.HybridCompute.Models;
+using Azure.ResourceManager.Models;
+
+namespace Azure.ResourceManager.HybridCompute
+{
+    /// <summary>
+    /// A class representing the Gateway data model.
+    /// Describes an Arc Gateway.
+    /// </summary>
+    public partial class GatewayData : TrackedResourceData
```
Comment (archerzz) [inline sdk/hybridcompute/Azure.ResourceManager.HybridCompute/src/Generated/GatewayData.cs:20]: The name is too generic. Consider appending RP prefix or renaming it.
- use `prepend-rp-prefix` to automatically append PR name. See example [here](https://github.com/Azure/azure-sdk-for-net/blob/176fd476588e8e61ebce747d66714cb39947cae5/sdk/azurestackhci/Azure.ResourceManager.Hci/src/autorest.md?plain=1#L52-L57).
- use `rename-mapping` to rename a model. See example [here](https://github.com/Azure/azure-sdk-for-net/blob/176fd476588e8e61ebce747d66714cb39947cae5/sdk/azurestackhci/Azure.ResourceManager.Hci/src/autorest.md?plain=1#L58-L76)

File: sdk/hybridcompute/Azure.ResourceManager.HybridCompute/src/Generated/GatewayData.cs [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+        internal GatewayData()
+        {
+        }
+
+        /// <summary> The provisioning state, which only appears in the response. </summary>
+        [WirePath("properties.provisioningState")]
+        public HybridComputeProvisioningState? ProvisioningState { get; }
+        /// <summary> A unique, immutable, identifier for the Gateway. </summary>
+        [WirePath("properties.gatewayId")]
+        public string GatewayId { get; }
```
Comment (archerzz) [inline sdk/hybridcompute/Azure.ResourceManager.HybridCompute/src/Generated/GatewayData.cs:94]: What's the format of this property value?
- If it's a GUID, use type [Guid](https://learn.microsoft.com/en-us/dotnet/api/system.guid?view=net-6.0). See example [here](https://github.com/Azure/azure-sdk-for-net/blob/bd953fc7a97bf617c892ce6314b50126b4021a4f/sdk/maps/Azure.ResourceManager.Maps/src/autorest.md?plain=1#L25).
- If it's a resource identifier, use type [ResourceIdentifier](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/Azure.Core/src/ResourceIdentifier.cs). See example [here](https://github.com/Azure/azure-sdk-for-net/blob/bd953fc7a97bf617c892ce6314b50126b4021a4f/sdk/search/Azure.Search.Documents/src/autorest.md?plain=1#L106-L115).

File: sdk/hybridcompute/Azure.ResourceManager.HybridCompute/src/Generated/Models/Disk.cs [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+
+#nullable disable
+
+using System;
+using System.Collections.Generic;
+
+namespace Azure.ResourceManager.HybridCompute.Models
+{
+    /// <summary> Describes a disk on the machine. </summary>
+    public partial class Disk
```
Comment (archerzz) [inline sdk/hybridcompute/Azure.ResourceManager.HybridCompute/src/Generated/Models/Disk.cs:14]: Name is too generic.

File: sdk/hybridcompute/Azure.ResourceManager.HybridCompute/src/Generated/Models/Disk.cs [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+
+        /// <summary> The path of the disk. </summary>
+        [WirePath("path")]
+        public string Path { get; }
+        /// <summary> The type of the disk. </summary>
+        [WirePath("diskType")]
+        public string DiskType { get; }
+        /// <summary> The generated ID of the disk. </summary>
+        [WirePath("generatedId")]
+        public string GeneratedId { get; }
```
Comment (archerzz) [inline sdk/hybridcompute/Azure.ResourceManager.HybridCompute/src/Generated/Models/Disk.cs:82]: What's the format of those IDs?

File: sdk/hybridcompute/Azure.ResourceManager.HybridCompute/src/Generated/Models/HardwareProfile.cs [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+
+#nullable disable
+
+using System;
+using System.Collections.Generic;
+
+namespace Azure.ResourceManager.HybridCompute.Models
+{
+    /// <summary> Describes the hardware of the machine. </summary>
+    public partial class HardwareProfile
```
Comment (archerzz) [inline sdk/hybridcompute/Azure.ResourceManager.HybridCompute/src/Generated/Models/HardwareProfile.cs:14]: Generic name

File: sdk/hybridcompute/Azure.ResourceManager.HybridCompute/src/Generated/Models/HybridComputeNetworkInterface.cs [hunk 6]
```diff
@@ -52,14 +52,29 @@ internal HybridComputeNetworkInterface()
...
             _serializedAdditionalRawData = serializedAdditionalRawData;
         }
 
+        /// <summary> Represents MAC address of the network interface. </summary>
+        [WirePath("macAddress")]
+        public string MacAddress { get; }
+        /// <summary> Represents the ID of the network interface. </summary>
+        [WirePath("id")]
+        public string Id { get; }
+        /// <summary> Represents the name of the network interface. </summary>
```
Comment (archerzz) [inline sdk/hybridcompute/Azure.ResourceManager.HybridCompute/src/Generated/Models/HybridComputeNetworkInterface.cs:75]: What's the format of the ID?

File: sdk/hybridcompute/Azure.ResourceManager.HybridCompute/src/Generated/Models/Processor.cs [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+
+#nullable disable
+
+using System;
+using System.Collections.Generic;
+
+namespace Azure.ResourceManager.HybridCompute.Models
+{
+    /// <summary> Describes the firmware of the machine. </summary>
+    public partial class Processor
```
Comment (archerzz) [inline sdk/hybridcompute/Azure.ResourceManager.HybridCompute/src/Generated/Models/Processor.cs:14]: Generic name

File: sdk/hybridcompute/Azure.ResourceManager.HybridCompute/src/Generated/Models/RunCommandManagedIdentity.cs [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+        internal RunCommandManagedIdentity(string clientId, string objectId, IDictionary<string, BinaryData> serializedAdditionalRawData)
+        {
+            ClientId = clientId;
+            ObjectId = objectId;
+            _serializedAdditionalRawData = serializedAdditionalRawData;
+        }
+
+        /// <summary> Client Id (GUID value) of the user-assigned managed identity. ObjectId should not be used if this is provided. </summary>
+        [WirePath("clientId")]
+        public string ClientId { get; set; }
```
Comment (archerzz) [inline sdk/hybridcompute/Azure.ResourceManager.HybridCompute/src/Generated/Models/RunCommandManagedIdentity.cs:66]: What's the format of those IDs?

File: sdk/hybridcompute/Azure.ResourceManager.HybridCompute/src/Generated/Models/Settings.cs [hunk 9]
```diff
@@ -0,0 +1,77 @@
...
+
+using System;
+using System.Collections.Generic;
+using Azure.Core;
+using Azure.ResourceManager.Models;
+
+namespace Azure.ResourceManager.HybridCompute.Models
+{
+    /// <summary> The Settings. </summary>
+    public partial class Settings : ResourceData
```
Comment (archerzz) [inline sdk/hybridcompute/Azure.ResourceManager.HybridCompute/src/Generated/Models/Settings.cs:16]: Generic name

File: sdk/hybridcompute/Azure.ResourceManager.HybridCompute/src/Generated/GatewayData.cs (hunk 1 above)
Comment (archerzz) [reply sdk/hybridcompute/Azure.ResourceManager.HybridCompute/src/Generated/GatewayData.cs:20]: same as `GatewayPatch` and `GatewayType` below

**Review by archerzz — CHANGES_REQUESTED** [review]
@yaotongms Need to update API signature. See the error logs for details. Thanks.

File: sdk/hybridcompute/Azure.ResourceManager.HybridCompute/src/Generated/Models/Settings.cs (hunk 9 above)
Comment (yaotongms) [reply sdk/hybridcompute/Azure.ResourceManager.HybridCompute/src/Generated/Models/Settings.cs:16]: fixed

File: sdk/hybridcompute/Azure.ResourceManager.HybridCompute/src/Generated/Models/Processor.cs (hunk 7 above)
Comment (yaotongms) [reply sdk/hybridcompute/Azure.ResourceManager.HybridCompute/src/Generated/Models/Processor.cs:14]: fixed

File: sdk/hybridcompute/Azure.ResourceManager.HybridCompute/src/Generated/Models/HardwareProfile.cs (hunk 5 above)
Comment (yaotongms) [reply sdk/hybridcompute/Azure.ResourceManager.HybridCompute/src/Generated/Models/HardwareProfile.cs:14]: fixed

File: sdk/hybridcompute/Azure.ResourceManager.HybridCompute/src/Generated/Models/Disk.cs (hunk 4 above)
Comment (yaotongms) [reply sdk/hybridcompute/Azure.ResourceManager.HybridCompute/src/Generated/Models/Disk.cs:82]: fixed

File: sdk/hybridcompute/Azure.ResourceManager.HybridCompute/src/Generated/Models/Disk.cs (hunk 3 above)
Comment (yaotongms) [reply sdk/hybridcompute/Azure.ResourceManager.HybridCompute/src/Generated/Models/Disk.cs:14]: fixed

File: sdk/hybridcompute/Azure.ResourceManager.HybridCompute/src/Generated/GatewayData.cs (hunk 1 above)
Comment (yaotongms) [reply sdk/hybridcompute/Azure.ResourceManager.HybridCompute/src/Generated/GatewayData.cs:20]: fixed

File: sdk/hybridcompute/Azure.ResourceManager.HybridCompute/src/Generated/Models/RunCommandManagedIdentity.cs (hunk 8 above)
Comment (yaotongms) [reply sdk/hybridcompute/Azure.ResourceManager.HybridCompute/src/Generated/Models/RunCommandManagedIdentity.cs:66]: fixed

File: sdk/hybridcompute/Azure.ResourceManager.HybridCompute/src/Generated/Models/Disk.cs (hunk 4 above)
Comment (yaotongms) [reply sdk/hybridcompute/Azure.ResourceManager.HybridCompute/src/Generated/Models/Disk.cs]: fixed

File: sdk/hybridcompute/Azure.ResourceManager.HybridCompute/src/Generated/GatewayData.cs (hunk 2 above)
Comment (yaotongms) [reply sdk/hybridcompute/Azure.ResourceManager.HybridCompute/src/Generated/GatewayData.cs]: fixed

File: sdk/hybridcompute/Azure.ResourceManager.HybridCompute/src/Generated/Models/HybridComputeNetworkInterface.cs (hunk 6 above)
Comment (yaotongms) [reply sdk/hybridcompute/Azure.ResourceManager.HybridCompute/src/Generated/Models/HybridComputeNetworkInterface.cs:75]: fixed

File: sdk/core/Azure.Core.TestFramework/tests/SessionRecords/RecordedTestBaseTests/WaitForCompletionErrorsArePropagated.json [hunk 10]
```diff
@@ -0,0 +1,4 @@
+{
```
Comment (jsquire) [inline sdk/core/Azure.Core.TestFramework/tests/SessionRecords/RecordedTestBaseTests/WaitForCompletionErrorsArePropagated.json:1]: The Azure.Core changes need to be removed.

**Review by jsquire — DISMISSED** [review]

File: sdk/core/Azure.Core.TestFramework/tests/SessionRecords/RecordedTestBaseTests/WaitForCompletionErrorsArePropagated.json (hunk 10 above)
Comment (yaotongms) [reply sdk/core/Azure.Core.TestFramework/tests/SessionRecords/RecordedTestBaseTests/WaitForCompletionErrorsArePropagated.json]: removed

Comment (yaotongms) [reply sdk/core/Azure.Core.TestFramework/tests/SessionRecords/RecordedTestBaseTests/WaitForCompletionErrorsArePropagated.json]: @jsquire I removed Azure.Core changes, but now I'm getting "Generated code is not up to date." error. 

running 'eng\scripts\CodeChecks.ps1 -ServiceDirectory hybridcompute' succeeds.

Comment (yaotongms) [reply sdk/core/Azure.Core.TestFramework/tests/SessionRecords/RecordedTestBaseTests/WaitForCompletionErrorsArePropagated.json]: updated

File: sdk/hybridcompute/Azure.ResourceManager.HybridCompute/src/Generated/Models/ExecutionState.cs [hunk 11]
```diff
@@ -0,0 +1,66 @@
...
+
+#nullable disable
+
+using System;
+using System.ComponentModel;
+
+namespace Azure.ResourceManager.HybridCompute.Models
+{
+    /// <summary> Script execution status. </summary>
+    public readonly partial struct ExecutionState : IEquatable<ExecutionState>
```
Comment (archerzz) [inline sdk/hybridcompute/Azure.ResourceManager.HybridCompute/src/Generated/Models/ExecutionState.cs:14]: The name is too generic. Consider appending RP prefix or renaming it.
- use `prepend-rp-prefix` to automatically append PR name. See example [here](https://github.com/Azure/azure-sdk-for-net/blob/176fd476588e8e61ebce747d66714cb39947cae5/sdk/azurestackhci/Azure.ResourceManager.Hci/src/autorest.md?plain=1#L52-L57).
- use `rename-mapping` to rename a model. See example [here](https://github.com/Azure/azure-sdk-for-net/blob/176fd476588e8e61ebce747d66714cb39947cae5/sdk/azurestackhci/Azure.ResourceManager.Hci/src/autorest.md?plain=1#L58-L76)

File: sdk/hybridcompute/Azure.ResourceManager.HybridCompute/src/Generated/Models/FirmwareProfile.cs [hunk 12]
```diff
@@ -0,0 +1,71 @@
...
+
+#nullable disable
+
+using System;
+using System.Collections.Generic;
+
+namespace Azure.ResourceManager.HybridCompute.Models
+{
+    /// <summary> Describes the firmware of the machine. </summary>
+    public partial class FirmwareProfile
```
Comment (archerzz) [inline sdk/hybridcompute/Azure.ResourceManager.HybridCompute/src/Generated/Models/FirmwareProfile.cs:14]: Generic name

**Review by archerzz — COMMENTED** [review]
Still some generic names to fix, Others look good to me.

If you are going to release a new version, don't forget to update `CHANGELOG.md`. See [here](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/78/Mgmt-.NET-SDK-release-process?anchor=signing-and-publishing-nuget-packges) for details.

File: sdk/hybridcompute/Azure.ResourceManager.HybridCompute/src/Generated/Models/FirmwareProfile.cs (hunk 12 above)
Comment (yaotongms) [reply sdk/hybridcompute/Azure.ResourceManager.HybridCompute/src/Generated/Models/FirmwareProfile.cs]: fixed

File: sdk/hybridcompute/Azure.ResourceManager.HybridCompute/src/Generated/Models/ExecutionState.cs (hunk 11 above)
Comment (yaotongms) [reply sdk/hybridcompute/Azure.ResourceManager.HybridCompute/src/Generated/Models/ExecutionState.cs]: fixed

File: sdk/hybridcompute/Azure.ResourceManager.HybridCompute/CHANGELOG.md [hunk 13]
```diff
@@ -1,14 +1,12 @@
 # Release History
 
-## 1.1.0-beta.1 (Unreleased)
+## 1.1.0 (2024-11-27)
```
Comment (ArthurMa1978) [inline sdk/hybridcompute/Azure.ResourceManager.HybridCompute/CHANGELOG.md:3]: The REST API version is preview so can't release a stable version of SDK

Comment (yaotongms) [reply sdk/hybridcompute/Azure.ResourceManager.HybridCompute/CHANGELOG.md]: fixed

**Review by archerzz — APPROVED** [review]

**Pull request merged**